src/swaps/fastbtc/FastbtcSwapProvider.js

```diff
@@ -0,0 +1,254 @@
+import { io } from 'socket.io-client'
+import BN from 'bignumber.js'
+import { mapValues } from 'lodash-es'
+import { SwapProvider } from '../SwapProvider'
+import { v4 as uuidv4 } from 'uuid'
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import { withInterval } from '../../store/actions/performNextAction/utils'
+import { prettyBalance } from '../../utils/coinFormatter'
+import cryptoassets from '@/utils/cryptoassets'
+import { getTxFee } from '../../utils/fees'
+
+const fastBtcSatoshiFee = 5000
+const fastBtcPercentageFee = 0.2
+
+class FastbtcSwapProvider extends SwapProvider {
+  constructor (config) {
+    super(config)
+    this.socketConnection = io(this.config.agent, {
+      reconnectionDelayMax: 10000
+    })
+
+    this.socketConnection.on('connect', function () {
+      console.log('FatBtc socket connected')
+    })
+
+    this.socketConnection.on('disconnect', function () {
+      console.log('FastBtc socket disconnected')
+    })
+  }
+
+  async getSupportedPairs () {
+    return []
+  }
+
+  async _getHistory (web3Addr) {
+    return new Promise((resolve, reject) => {
+      this.socketConnection.emit('getDepositHistory', web3Addr, (res) => {
+        if (res && res.error) {
+          reject(res.err)
+        }
+        resolve(res)
+      })
+    })
+  }
+
+  async _getAddress (web3Addr) {
+    return new Promise((resolve, reject) => {
+      this.socketConnection.emit('getDepositAddress', web3Addr, (err, res) => {
+        if (err) {
+          reject(err)
+        }
+        resolve(res)
+      })
+    })
+  }
+
+  async _getTxAmount () {
+    return new Promise((resolve, reject) => {
+      this.socketConnection.emit('txAmount', (res) => {
+        resolve(res)
+      })
+    })
+  }
+
+  async getQuote ({ network, from, to, amount }) {
+    const fromAmountInUnit = BN(currencyToUnit(cryptoassets[from], BN(amount)))
+    const validAmountRange = await this._getTxAmount()
+    const isQuoteAmountInTheRange = amount <= validAmountRange.max && amount >= validAmountRange.min
+    if (from !== 'BTC' || to !== 'RBTC' || !isQuoteAmountInTheRange) return null
```

Comment:
from and to can be checked at the beginning to avoid any other calls if not needed